src/middleware/data-factory.ts

```diff
@@ -1,14 +1,13 @@
 import {
-  DefaultContext, DefaultState, Middleware, Next,
+  DefaultContextExtends, DefaultStateExtends, Middleware, Next,
 } from 'koa';
 import { DataFactory } from 'rdf-js';
 
-export type DataFactoryContext<Context extends DefaultContext = DefaultContext,
+export type DataFactoryContext<Context extends DefaultContextExtends = DefaultContextExtends,
   Factory extends DataFactory = DataFactory> = Context & { dataFactory: Factory };
 
-export default <Factory extends DataFactory>(dataFactory: Factory):
-  Middleware<DefaultState, DataFactoryContext<{}, Factory>> => (
-  async (context: DataFactoryContext<{}, Factory>, next: Next): Promise<void> => {
+export default (dataFactory: DataFactory): Middleware<DefaultStateExtends, DefaultContextExtends> => (
+  async (context: DefaultContextExtends, next: Next): Promise<void> => {
```

Comment:
This is like https://github.com/libero/article-store/pull/115#discussion_r363821502, the `dataFactory` context hasn't been set yet so using `DataFactoryContext` isn't right.